core/adapters/http_test.go

```diff
@@ -284,23 +262,307 @@ func TestQueryParameters(t *testing.T) {
 				URL:         cltest.WebURL(t, test.startingUrl),
 				QueryParams: qp,
 			}
-			if test.wantErrored {
-				requestGET, _ := hga.GetRequest()
-				assert.Equal(t, test.expectedURL, requestGET.URL.String())
-				assert.Equal(t, test.expectedParams, hga.QueryParams)
-				requestPOST, _ := hpa.GetRequest("")
-				assert.Equal(t, test.expectedURL, requestPOST.URL.String())
-				assert.Equal(t, test.expectedParams, hpa.QueryParams)
-				assert.NotNil(t, err)
-			} else {
-				requestGET, _ := hga.GetRequest()
-				assert.Equal(t, test.expectedURL, requestGET.URL.String())
-				assert.Equal(t, test.expectedParams, hga.QueryParams)
-				requestPOST, _ := hpa.GetRequest("")
-				assert.Equal(t, test.expectedURL, requestPOST.URL.String())
-				assert.Equal(t, test.expectedParams, hpa.QueryParams)
-				assert.Nil(t, err)
+			requestGET, _ := hga.GetRequest()
+			assert.Equal(t, test.expectedURL, requestGET.URL.String())
+			assert.Equal(t, test.expectedParams, hga.QueryParams)
+			requestPOST, _ := hpa.GetRequest("")
+			assert.Equal(t, test.expectedURL, requestPOST.URL.String())
+			assert.Equal(t, test.expectedParams, hpa.QueryParams)
+			assert.Nil(t, err)
+		})
+	}
+}
+
+func TestQueryParameters_Error(t *testing.T) {
+	t.Parallel()
+
+	baseUrl := "http://example.com"
+
+	cases := []struct {
+		name           string
+		queryParams    string
+		startingUrl    string
+		expectedParams adapters.QueryParameters
+		expectedURL    string
+	}{
+		{
+			"odd number of params",
+			`["firstKey","firstVal","secondKey","secondVal","bad"]`,
+			baseUrl,
+			adapters.QueryParameters{},
+			baseUrl,
+		},
+		{
+			"bad format of string",
+			`"firstKey=firstVal&secondKey=secondVal&bad"`,
+			baseUrl,
+			adapters.QueryParameters{},
+			baseUrl,
+		},
+	}
+
+	for _, tt := range cases {
+		test := tt
```

Comment:
We no longer need this I believe since the subtests are not run in parallel. ie `t.Parallel()` is up top, not inside this `t.Run(...)`. Let's clean it up in all tests.

```
for _, test := range cases {
        t.Run(test.name, func(t *testing.T) {
}
}
```